airflow/contrib/operators/aws_athena_operator.py

```diff
@@ -0,0 +1,97 @@
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+from airflow.contrib.hooks.aws_athena_hook import AWSAthenaHook
+
+
+class AWSAthenaOperator(BaseOperator):
+    """
+    Airflow operator to run presto queries on athena.
+
+    :param query: Presto to be run on athena. (templated)
+    :type query: str
+    :param database: Database to select. (templated)
+    :type database: str
+    :param output_location: s3 path to write the query results into. (templated)
+    :type output_location: str
+    :param aws_conn_id: aws connection to use.
+    :type aws_conn_id: str
+    """
+
+    ui_color = '#44b5e2'
+    template_fields = ('query', 'database', 'output_location')
+
+    @apply_defaults
+    def __init__(self, query, database, output_location, aws_conn_id='aws_default', *args, **kwargs):
+        super(AWSAthenaOperator, self).__init__(*args, **kwargs)
+        self.query = query
+        self.database = database
+        self.output_location = output_location
+        self.aws_conn_id = aws_conn_id
+        self.client_request_token = kwargs.get('client_request_token')
+        self.query_execution_context = kwargs.get('query_execution_context') or {}
+        self.result_configuration = kwargs.get('result_configuration') or {}
+        self.query_execution_context['Database'] = self.database
+        self.result_configuration['OutputLocation'] = self.output_location
+        self.query_execution_id = None
+        self.hook = None
+
+    def get_hook(self):
+        return AWSAthenaHook(self.aws_conn_id)
+
+    def pre_execute(self, context):
+        """
+        Create AWS Athena hook and a boto3 athena client
+        """
+        self.hook = self.get_hook()
+        self.hook.get_conn()
+
+    def execute(self, context):
+        """
+        Run Presto Query on Athena
+        """
+        self.query_execution_id = self.hook.run_query(self.query, self.query_execution_context,
+                                                      self.result_configuration, self.client_request_token)
+
+    def post_execute(self, context):
+        """
+        Poll submitted athena query until query will reach of following states (SUCCESS, FAILED, CANCELLED)
+        """
+        self.hook.poll_query_status(self.query_execution_id)
```

Comment:
The pre_ and post_ execute functions should be all merged in to execute - for pre it doesn't matter, but post in particular is the bulk of the time, and execution_timeouts are only applied to `execute()` so that needs to be where we do the waiting.